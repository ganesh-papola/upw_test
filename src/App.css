

p,
h1,
h2,
h3,
h4,
h5,
h6 {
  margin: 0 !important;
}

input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

/* Firefox */
input[type="number"] {
  -moz-appearance: textfield;
}
