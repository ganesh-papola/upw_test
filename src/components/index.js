export * from "./common";
export * from "./Hoc";
