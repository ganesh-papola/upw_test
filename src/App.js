import "./App.css";
import { Provider } from "react-redux";
import { persistor, store } from "./redux/store";
import Routes from "./routes/routes";
import { PersistGate } from "redux-persist/integration/react";
import { HelmetProvider } from "react-helmet-async";
import { LoaderHoc } from "./components";

function App() {
  return (
    <Provider store={store}>
      <PersistGate persistor={persistor}>
        <HelmetProvider>
          <LoaderHoc>
            <Routes />
          </LoaderHoc>
        </HelmetProvider>
      </PersistGate>
    </Provider>
  );
}

export default App;
