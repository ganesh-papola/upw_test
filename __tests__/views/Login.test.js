import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Login } from "../../src/views"
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

jest.mock('react-chartjs-2', () => 'Chart')
  jest.mock('chart.js', () => ({
    register: () => null
  }));
  const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const mockInitialState = {
    auth: {},
    common: {}
}
const store = mockStore(mockInitialState);

describe('Dashboard test', () => {
    const props = {}
    it('this is components snapshot rendering test', () => {
        const { container } = render(
            <Provider store={store}>
                <Login {...props} />
          </Provider>
        );
        expect(container).toMatchSnapshot();
      });
})
