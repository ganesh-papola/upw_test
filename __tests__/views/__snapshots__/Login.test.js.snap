// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard test this is components snapshot rendering test 1`] = `
<div>
  <section
    class="login"
  >
    <section
      class="formSection"
    >
      <div
        class="headingSection"
      >
        <label
          class="title"
        >
          Sign in
        </label>
      </div>
      <div
        class="formContainer"
      >
        <div
          class="formSubContainer"
        >
          <form
            action="#"
          >
            <div
              class="inputContainer loginInputContainer"
            >
              <div
                class="formGroup"
              >
                <input
                  class="formField"
                  id="login__0"
                  max="100"
                  min="10"
                  name="Username"
                  placeholder="Username"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="inputContainerLabelDiv"
              >
                <span
                  class="inputContainerError"
                />
              </div>
            </div>
            <div
              class="inputContainer loginInputContainer"
            >
              <div
                class="formGroup"
              >
                <input
                  class="formField"
                  id="login__1"
                  max="100"
                  min="10"
                  name="Password"
                  placeholder="Password"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="inputContainerLabelDiv"
              >
                <span
                  class="inputContainerError"
                />
              </div>
            </div>
            <div
              class="buttonContainer"
            >
              <button
                class="button"
                type="submit"
              >
                Sign in
              </button>
            </div>
          </form>
        </div>
      </div>
    </section>
  </section>
</div>
`;
